### PR TITLE
Generate TypeScript Definitions

### PR DESCRIPTION
**Pull request by cravend**
This PR adds TypeScript definitions to the Hack Club theme. No functionality changed, running `yarn prepare` automatically compiles the TypeScript files.

I also updated the dependencies because I couldn't get it to run.

### CODE REVIEW

**Comment by kenmueller** [comment]
Looks great, thanks for showing me `"declaration": true`!

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[submodule "docs/workshops"]
+	url = https://github.com/hackclub/workshops
```
Comment (lachlanjc) [inline .gitmodules:2]: Sorry if I'm missing something, what's this here for?

File: packages/theme/package.json [hunk 2]
```diff
@@ -7,9 +7,10 @@
   "main": "dist/index.js",
   "module": "dist/index.esm.js",
   "unpkg": "dist/index.umd.js",
+  "types": "dist/index.d.ts",
   "sideEffects": false,
   "scripts": {
-    "prepare": "rm -rf ./dist && microbundle",
+    "prepare": "rm -rf ./dist && tsc --project tsconfig.json && microbundle",
```
Comment (lachlanjc) [inline packages/theme/package.json:13]: I haven't investigated this, does Microbundle's built-in TypeScript not work here? https://github.com/developit/microbundle#using-with-typescript

File: packages/theme/src/index.ts [hunk 3]
```diff
@@ -340,7 +340,15 @@ const theme = {
       my: 4,
       borderCollapse: 'separate',
       borderSpacing: 0,
-      [['th', 'td']]: {
+      th: {
```
Comment (lachlanjc) [inline packages/theme/src/index.ts:343]: Let's avoid duplicating code here. Does `'th,td'` work instead?

File: .gitmodules (hunk 1 above)
Comment (cravend) [reply .gitmodules:2]: Oh, I thought that `docs/workshops` was a submodule? It was showing up as invalid when I cloned it (idk if it's pointing to a private repo?) so I just linked it.

File: packages/theme/src/index.ts (hunk 3 above)
Comment (cravend) [reply packages/theme/src/index.ts:343]: By TypeScript's standards, yes. Does it work for `theme-ui`? unknown...

File: packages/theme/package.json (hunk 2 above)
Comment (cravend) [reply packages/theme/package.json]: Yeah it does! Updated :-)

File: packages/theme/src/index.ts (hunk 3 above)
Comment (cravend) [reply packages/theme/src/index.ts:343]: Actually I'm not convinced that `[['th', 'td']]` does what it we think it does. That's just going to compute to `'th,td'` which doesn't work w/theme-ui. I just tried it with my own theme (I don't have a table to test it with).

Comment (lachlanjc) [reply packages/theme/src/index.ts:343]: It should work with Theme UI—I use selectors like that all the time

File: .gitmodules (hunk 1 above)
Comment (lachlanjc) [reply .gitmodules:2]: There shouldn’t be any submodules here. If you’re using/developing Theme locally, you’ll want to use `yarn link`, or in production, just install the package…what made you think there’s a submodule?

Comment (cravend) [reply .gitmodules:2]: GitKraken yelled at me about an invalid submodule 😞 also [this](https://github.com/hackclub/theme/commit/e36dde335cc7ae2d74cbdaa3453dffe0c6c6fe15#diff-1b877e76739b502b9bd616bb41b4a7a8). However, it does not at all affect what I'm doing, so I can get rid of it.

**Comment by lachlanjc** [comment]
I’m happy to release this but unfortunately I’ve been sitting on a local branch making huge changes to Theme & adding a bunch of helper functions & variants & colors & such…I’ll probably just convert all those to TS next week 😅

File: packages/theme/src/index.ts (hunk 3 above)
Comment (cravend) [reply packages/theme/src/index.ts:343]: Interesting, that's good to know! Updated.

**Comment by cravend** [comment]
> I’m happy to release this but unfortunately I’ve been sitting on a local branch making huge changes to Theme & adding a bunch of helper functions & variants & colors & such…I’ll probably just convert all those to TS next week 😅

Exciting stuff! If you want a beta tester/TS user, let me know.

File: .gitmodules (hunk 1 above)
Comment (lachlanjc) [reply .gitmodules]: Ooh, not sure why that’s there, thanks for pointing out

**Pull request merged**